### DISCUSSION TITLE
how to stop text reorganization?

### DISCUSSION THREAD

**Discussion by wolfindark** [OpenModelica/OpenModelica#10018] [Q&A] [answered]
Hi, 
I organize the text in OMEdit, if I add a new component, whole text automatically reorganized into each component per in a single line. All indentations and newlines disappers. 
I am sure many people asked about it, but I could not find any answer. 

thanks

**Comment by casella** [comment, marked as answer]
OMEdit is not really at text editor, it continuously interacts with the OMC parser in the background. Normally, indentation and formatting should be preserved, but in some cases this fails, and then OMC returns pretty-printed code where the original formatting is lost. Most causes of such failures have been fixed, but there are probably still some out there.

It would be great if you could [open a ticket](https://github.com/OpenModelica/OpenModelica/issues/new?assignees=&labels=&template=bug_report.md&title=) where you post the code so that we can reproduce the issue and fix it for good.

**Reply by wolfindark** [reply]
Thanks forthe answer. 
It happend when I added System component from the Fluid library by drag and drop onto diagram view... 
I will opean a ticket. 
I appreciate your effort. Thanks again.

**Reply by casella** [reply]
No problem :)

**Comment by mpradovelasco** [comment]
Hi,

I have read this issue with detail and I suppose that it is a repeated comment or subject of OMEdit users. However, I would wish only to emphasize that in my opinion this behaviour affects strongly the usability of the editor. It seems to have been improved along the last versions, however the event still appears, for instance, when the icon of a class inside a .mo package is modified (not always, but many times). 

When the event occurs (the message `Scripting Warning Could not preserve the formatting of the model instead internal pretty-printing algorithm is used.` appears), in large .mo files, I prefer to make an external copy of the .mo file, before it will be modify by OMEdit and use it to recover the format (adding the code associated to the modification that promoted the event). 

Thank you very much again for all your effort!

**Comment by casella** [comment]
I suggest two strategies to cope with this:

1. Avoid large .mo files. This can be done by splitting packages and sub-packages into separate files, rather than keeping them in single files. You can get down to the point that each class is contained in a single file. This makes the interaction between OMEdit and OMC much faster, and limits the reorganization damage to that class in the worst case.

2. If you experience this kind of issue, please open tickets with indications on how to reproduce it, so we can fix the diff algorithm and improve it.

I'm afraid there's not many other options 😅

**Comment by mpradovelasco** [comment]
Thank you for your suggestions! 
I have opened a ticket for a very frequent case (in my experience) (#10715).
